packages/core/src/NavigationFocusedContext.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { NavigationHelpers, ParamListBase } from './types';
+
+export type FocusedNavigationCallback = (
+  navigation: NavigationHelpers<ParamListBase>
+) => any;
+
+export type FocusedNavigationListener = (
+  callback: FocusedNavigationCallback
+) => { handled: boolean; result: any };
+
+const MISSING_CONTEXT_ERROR =
+  "We couldn't find a navigation focused context. Have you wrapped your app with 'NavigationContainer'?";
+
+/**
+ * Context which holds the required helpers needed to access focused navigator.
+ */
+
+const NavigationFocusedContext = React.createContext<{
+  addNavigationListener: (listener: FocusedNavigationListener) => void;
+}>({
+  get addNavigationListener(): any {
```

Comment:
Not sure about `addNavigationListener` since we're not listening to navigation. Maybe something like `addFocusedListener` to match the naming of the context and the hook?